module/polls/helper.js

```diff
@@ -0,0 +1,745 @@
+                          form: pollCreationForm,
+                          emoticons: emoticons,
+                          gestures: gestures
+                    }
+                });
+
+            } catch (error) {
+                 return resolve({
+                    success: false,
+                    message: error.message,
+                    data: false
+                });
+            }
+        });
+    }
+
+    /**
+     * Create poll
+     * @method
+     * @name create
+     * @param {Object} pollData - poll creation data
+     * @param {String} userId - userId
+     * @param {String} appName - Name of the app
+     * @returns {String} - Sharable link .
+     */
+
+    static create(pollData= {}, userId= "", appName= "") {
+        return new Promise(async (resolve, reject) => {
+            try {
+
+                if (Object.keys(pollData).length == 0) {
+                    throw new Error (messageConstants.apiResponses.POLL_DATA_REQUIRED);
+                }
+
+                if (userId == "") {
+                    throw new Error (messageConstants.apiResponses.USER_ID_REQUIRED_CHECK);
+                }
+
+                if (appName == "") {
+                    throw new Error (messageConstants.apiResponses.APP_NAME_FIELD_REQUIRED);
+                }
+                
+                let pollsModel = Object.keys(pollsSchema.schema);
+                
+                let pollMetainformation = {};
+
+                Object.keys(pollData).forEach( key => {
+                    if(!pollsModel.includes(key)) {
+                        pollMetainformation[key] = pollData[key];
+                    }
+                })
+               
+                let pollDocument = {
+                    name: pollData.name,
+                    creator: pollData.creator,
+                    createdBy: userId,
+                    startDate: new Date(),
+                    endDate: new Date(new Date().setDate(new Date().getDate() + pollData.endDate)),
+                    metaInformation: pollMetainformation,
+                    isDeleted: false,
+                    status: "active",
+                    numberOfResponses: 0
+                }
+                
+                let questionArray = [];
+                let result = {};
+
+                for(let question=0; question< pollData.questions.length; question++) {
+   
+                    let options = [];
+                    pollData.questions[question].qid = gen.utils.generateUUId();;
+                    result[pollData.questions[question].qid] = {}
+               
+                    if (pollData.questions[question].options.length > 0) {
+                       let i = 1;
+               
+                       for(let option=0; option< pollData.questions[question].options.length; option++) {
+                           options.push(
+                                          {
+                                            value: "R" + i,
```

Comment:
@deepakr123 - Instead of i, just user option+1

Reply:
resolved

Reply:
resolved